modular_skyrat/code/modules/tidezombies/machinery/mysterycrate.dm

```diff
@@ -0,0 +1,187 @@
+		/obj/item/gun/energy/pulse = 1,
+		/obj/item/gun/syringe = 25,
+		/obj/item/gun/syringe/rapidsyringe = 20,
+		/obj/item/gun/syringe/dna = 20,
+		/obj/item/gun/syringe/syndicate = 15,
+		/obj/item/gun/magic/wand/resurrection = 1,
+		/obj/item/gun/magic/wand/fireball = 5,
+		/obj/item/gun/magic/staff/honk = 5,
+	)
+	//Possible weapons associated with their icon and sprite.
+	//Generated on spawn, should not worry too much, though this does mean
+	//spawning these is quite a hefty toll.
+	var/list/possible_icons = list()
+	//The current image overlay for the funny crate
+	var/image/item_overlay = null
+
+/obj/structure/closet/crate/mysterycrate/Initialize(mapload)
+	. = ..()
+	for(var/i in possible_outcomes)
+		var/obj/item/I = new i()
+		possible_icons[i] = list(I.icon, I.icon_state)
+		qdel(I)
+	generate_images()
+
+/obj/structure/closet/crate/mysterycrate/examine(mob/user)
+	. = ..()
+	. += "<span class='warning'>\The [src] has <b>[uses]</b> uses left, each use costing <b>[cost_per_gun]</b> credits.</span>"
+
+/obj/structure/closet/crate/mysterycrate/update_icon()
+	. = ..()
+	cut_overlays()
+	if(item_overlay)
+		add_overlay(item_overlay)
+
+/obj/structure/closet/crate/mysterycrate/proc/generate_images()
+	for(var/i in possible_icons)
+		var/image/eemage = image(possible_icons[i][1], src, possible_icons[i][2], layer, SOUTH)
+		eemage.pixel_x = 6
+		possible_icons[i] = eemage
+
+/obj/structure/closet/crate/mysterycrate/attack_hand(mob/user)
+	if(uses <= 0)
+		..()
+	else
+		if(!in_use && (current_cash >= cost_per_gun) && uses)
+			current_cash -= cost_per_gun
+			roll_for_gun()
+		else if(!in_use && uses)
+			to_chat(user, "<span class='warning'>\The [src] needs at least [cost_per_gun] credits to roll for an item!</span>")
+		else
+			to_chat(user, "<span class='warning'>\The [src] is rolling for an item already!</span>")
+
+/obj/structure/closet/crate/mysterycrate/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/holochip))
+		var/obj/item/holochip/H = W
+		current_cash += H.credits
+		qdel(H)
+		playsound(src, kaching, 100, 0)
+	else if(istype(W, /obj/item/stack/spacecash))
+		var/obj/item/stack/spacecash/S = W
+		current_cash += S.value
+		qdel(S)
+		playsound(src, kaching, 100, 0)
+	else
+		..()
+
+/obj/structure/closet/crate/mysterycrate/proc/roll_for_gun()
+	uses--
+	in_use = TRUE
+	var/randumb
+	icon_state = "mysteryplushopen"
+	opened = TRUE
+	playsound(src, animation_jingle, 100, 0)
+	update_icon()
+	for(var/i in 1 to 35)
+		randumb = pickweight(possible_outcomes, 1)
+		item_overlay = possible_icons[randumb]
+		update_icon()
+		sleep(2)
+	icon_state = initial(icon_state)
+	opened = FALSE
+	item_overlay = null
+	in_use = FALSE
+	update_icon()
+	var/turf/T = get_step(src, SOUTH)
+	new randumb(T)
+	if(!uses)
+		QDEL_IN(src, 6 SECONDS)
+		playsound(src, byebye, 100, 0)
+
+/obj/structure/closet/crate/mysterycrate/Destroy()
+	. = ..()
+	new /obj/item/toy/plush/mysterycrate(src.loc)
+
+/obj/item/toy/plush/mysterycrate
```

Comment:
?

Reply:
the plush gets left behind when the crate gets deleted that is the meme

Reply:
Is this someone's OC?